### ISSUE TITLE
Question: bind text inputs to the right of the tree-view

### ISSUE THREAD

**Issue by rajush** [iVantage/angular-ivh-treeview#41]
Hello! First-off we love this module. We were hoping you could help us.

We are looking to bind text inputs to the right of the tree-view. 

We are currently using the beta with custom templates and have what you can see in the image linked but we would like to have them behave or be data bound with the nodes they are attached to.

We have been struggling with this for a while now. We were hoping you could shed some light on the best way to accomplish this.

Here’s the image:

http://i.imgur.com/bvaPUjj.png

Thank you!

**Comment by jtrussell** [comment]
Are you binding to a property on the tree nodes themselves? You could do something like this: http://jsbin.com/duqaxeqelo/1/edit?html,js,output

Right now it's not super well documented but you have access to two variables on the template scope: `node` and `ctrl` which are the node itself and the [treeview controller](https://github.com/iVantage/angular-ivh-treeview/blob/master/src/scripts/directives/ivh-treeview.js#L53) respectively.

As a general disclaimer I'll just say that this branch is still pretty rough, how we expose certain things may change (i.e. there may be something like an `ivh-treeview-node-as` attribute eventually) Any feedback would be appreciated!

**Comment by rajush** [comment]
Thanks for your response. Really appreciated it. Looked at the demo and it did help a bit for now.
So, currently, we are auto generating the labels and children (nested loop) based  on the object model that we get back on the service call.
````
tree = [{
            label: ' ', 
            children: []
        }];
````

With the help of your custom template model, we were able to bind the data for each node from the object model, to the custom input field. The next thing would be to be able to read the data in the input box in other controllers. Hopefully, it could be done without actually tampering with the source code..(will keep you posted!).

Might need to revisit you soon. Thanks.

**Comment by rajush** [comment]
So we were able to bind custom data to input box for each node. Is there a way to make the input box act the exact same way like the check-boxes? So, if the text in the input field on the parent node is changed, then it gets carried over all the way down to the children nodes, but if the text is changed only in the child node, then it will be only for that particular child node.

Any suggestion that you might have would be highly appreciated, and the custom template, even though it's in beta, already rocks for us! Thank you!

**Comment by jtrussell** [comment]
Internally the directive uses the [ivhTreeviewBfs service](https://github.com/iVantage/angular-ivh-treeview#tree-traversal) to do tree traversal - e.g. for updating checkboxes. You could use the same service to update this text for child nodes:

```
ivhTreeviewBfs(nodeWithChangedText, function(child) {
  child.specialText = nodeWithChangedText.specialText;
});
```

**Comment by rajush** [comment]
I tried to apply but did not quite get this to work.
Also,  ng-blur does not trigger when I use it on <input> tag of the custom template . Might there be any reason for this behavior? What I'm trying to do is validate the value in the input box on blur that are attached on each node.

**Comment by jtrussell** [comment]
Mind providing an example with that jsbin template of what you tried?

Is `ng-blur` attached to the tree node or the input element? I think I'd start with something along these lines: http://jsbin.com/jotiqa/1/edit

**Comment by rajush** [comment]
That was exactly what I need. Thank you!!! I was missing the link function.

I also updated the example a bit. So what if I want to update the input box of all children with the same value that I enter in the input box in parent node? Like, carry the value down to all the children if the checkbox is checked on the parent node. You did mention something similar above, but couldn't quite get it to work.

+ Hats (parent input value)
       - Flat cap (parent input value)
           - Large (parent input value)
       - Fedora (parent input value)
           - Large (parent input value)
+ Pens 
      - Fountain (parent input value)
          - Small (parent input value)
      - Ballpoint 
           - Small

http://jsbin.com/xupusuzahe/1/edit?html,js,console,output

**Comment by jtrussell** [comment]
Something like this? http://jsbin.com/jotiqa/2/edit

**Comment by rajush** [comment]
That works! I should be able to tweak it here and there.Thank you !

**Comment by rajush** [comment]
Is there a way to check the value in the parent node's input box, after some value has been entered in the children's input box. Basically, what I'm trying to achieve is check from the lowest child level that if there are any values in the input field of it's parent node.

Thank you.

**Comment by jtrussell** [comment]
There's really nothing baked in to do that sort of validation, you can use `ivhTreeviewBfs` to find your node and it'll also give you a list of that node's parents but it would probably be more efficient to roll your solution specific to your needs. For example, you might want to have each node maintain a reference to its parent so you can easily walk up the tree. But yeah, sorry, it's definitely doable with the tools here but they aren't really optimized for what you're trying to do.

**Comment by jtrussell** [comment]
Feel free to reopen if needed.

**Comment by rajush** [comment]
Just a quick question here, on the above example that you provided,
  http://jsbin.com/jotiqa/2/edit

to use the $templateCache.get('node-template.html'), does it have to be under

<script> tag with "id" in it? Is there a way to point it to it's own external .html template, instead of providing inline template like 

<pre>nodeTpl: [
      '<div title="{{ctrl.label(node)}}">',
      '</div>'
    ].join('\n')
</pre>

used in <code>ivhTreeviewOptionsProvider.set({})</code>.

In short, is there a way for that <code>nodeTpl:[ ]</code> to load html template?

Thanks in advance if you ever get a chance to come back at this.

**Comment by jtrussell** [comment]
At present there does not exist something like `node-tpl-url`, it would be nice and I could get behind adding it.

The best you can do right now (a bit verbose but not so bad IMO) is to use `ng-include` in a transcluded template:

```html
<div ivh-treeview="fancy.stuff">
  <script type="text/ng-template">
    <div ng-include="'path/to/node-tpl.html'"></div>
  </script>
</div>
```

Here's the same demo updated to use this approach:

http://jsbin.com/malako/edit?html,js,output

**Comment by rajush** [comment]
Thank you for this. So it seems like the template translusion works only work with you new version. I was using the older version, not sure which exactly is that, but to bind the data, it uses
 `{{ctrl.label(node)}}`.
Now, it says 
 `{{trvw.label(node)}}`,
and works as expected.

So, currently pointing to the remote version from rawgit. For local copy, does the latest built version works the same or there is a specific version that I should be getting. 
Thanks once again.

**Comment by jtrussell** [comment]
v1.x will use `trvw.label(node)`. The latest build is 1.0.1 so you should be good with `trvw.*`.

Looks like we neglected to add a note for this in the migration notes, sorry about that.

(Created #91)
